aravisApp/src/ADAravis.cpp

```diff
@@ -670,6 +677,22 @@ asynStatus ADAravis::allocBuffer() {
     return asynSuccess;
 }
 
+void ADAravis::reconnectIfNeeded() {
+    int autoConnect;
+    if (this->connectionValid == 0) {
+        getIntegerParam(AravisAutoConnect, &autoConnect);
+        if (autoConnect) {
+            /* lock to make sure the main thread doesn't connect at the
+             * same time via the reset PV */
+            this->lock();
```

Comment:
Isn't it unsafe to call `getIntegerParam` without the lock being held? A PV write could have triggered a write to the paramList at the same time.

Reply:
Thanks for catching that